File: Serene/Serene.Script/Imports/FormContexts/Administration.RoleForm.cs

namespace Serene.Administration
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Runtime.CompilerServices;

    public partial class RoleForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Administration.Role";
    
        public RoleForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor RoleName { [InlineCode("{this}.w('RoleName', Serenity.StringEditor)")] get; private set; }
    }
}

File: Serene/Serene.Web/Modules/Common/Imports/ServerTypings/Administration.RoleForm.ts
namespace Serene.Administration {
    export class RoleForm extends Serenity.PrefixedContext {
        static formKey = 'Administration.Role';

    }

    export interface RoleForm {
        RoleName: Serenity.StringEditor;
    }

    [['RoleName', () => Serenity.StringEditor]].forEach(x => Object.defineProperty(RoleForm.prototype, <string>x[0], { get: function () { return this.w(x[0], (x[1] as any)()); }, enumerable: true, configurable: true }));
}

